src/main/scala/shine/OpenCL/HostCodeGenerator.scala

```diff
@@ -0,0 +1,200 @@
+package shine.OpenCL
+
+import arithexpr.arithmetic
+import shine.C
+import shine.C.AST._
+import shine.DPIA.Phrases._
+import shine.DPIA.Types._
+import shine.DPIA.VarType
+import shine.OpenCL.compilation.HostManagedBuffers
+import shine.OpenCL.primitives.imperative.{KernelCallCmd, NewManagedBuffer}
+
+import scala.collection.{immutable, mutable}
+
+object HostCodeGenerator {
+  def apply(): HostCodeGenerator =
+    new HostCodeGenerator(mutable.ListBuffer[Decl](), immutable.Map[String, arithmetic.Range]())
+}
+
+case class HostCodeGenerator(override val decls: C.CodeGenerator.Declarations,
+                             override val ranges: C.CodeGenerator.Ranges)
+  extends C.CodeGenerator(decls, ranges)
+{
+  override def name: String = "OpenCL Host"
+
+  override def cmd(env: Environment): Phrase[CommType] => Stmt = {
+    case KernelCallCmd(name, LocalSize(ls), GlobalSize(gs), output, args) =>
```

Comment:
maybe these cases are soo big that it makes sense to push them into individual functions?

Reply:
I can do that, although I am not convinced that it would be more readable.

Reply:
At least it helps to avoid deep indentation

Reply:
doing it